src/components/ShowMenus/ShowMenus.js

```diff
@@ -1,15 +1,23 @@
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 import styled from 'styled-components';
 
-import menusdata from '../../data.js';
+export default function ShowMenus({menus, curClickedCategory}) {
+  const [renderMenus, setRenderMenus] = useState([]);
 
-export default function ShowMenus() {
-  /* eslint-disable-next-line */
-  const [menus, setmenus] = useState(menusdata);
+  useEffect(() => {
+    if (curClickedCategory === 'All') setRenderMenus(menus);
+    else
+      setRenderMenus(
+        menus.filter(menu => {
+          return curClickedCategory === menu.category;
+        })
+      );
+    /* eslint-disable-next-line */
```

Comment:
Please delete this eslint-disable. Try to fix the error instead. Maybe useEffect is not the best solution for this case.

Reply:
```
  <MenusContainer>
          {menus.map(menu => {
          return curClickedCategory === menu.category ? (
           showMenu(menu)
          ) : curClickedCategory === 'All' ? (
           showMenu(menu)
          ) : (
           <></>
          );
       })}
    </MenusContainer>
```
At first I have a conditional render like this, the problem, it gives a error:  no unique key.  This means I have to give this thing <></>  a unique key, but to give this a key, it has to be div or some real html element, empty divs effect the layout.
Do you have any better idea?
By the way, I need help, not just this is wrong that is wrong.

Reply:
You need just the first else case. Either you have chosen a category or all will be displayed.

Reply:
> 

The problem is, only in map function I can know if a menu should be rendered or not, and map function should be always return something, even if I don't want a menu to be rendered, I have to return something, that means <></>, if I return this,  I need a unique key, to give a unique key, this thing have to be divs.

Reply:
> You need just the first else case. Either you have chosen a category or all will be displayed.

Another idea,  I can do it when onClick happens, I'll let you know when I finish.

Reply:
Is there a case you dont want to render anything?

Reply:
Hi Stefan, I just used filter and then map to solve the problem, it works good, less codes. Thank you very much!